aptos-move/framework/supra-framework/sources/multisig_voting.spec.move

```diff
@@ -0,0 +1,427 @@
+        aborts_if len(execution_hash) == 0;
+        let execution_key = std::string::spec_utf8(IS_MULTI_STEP_PROPOSAL_KEY);
+        aborts_if simple_map::spec_contains_key(metadata, execution_key);
+        aborts_if voting_forum.next_proposal_id + 1 > MAX_U64;
+        let is_multi_step_in_execution_key = std::string::spec_utf8(IS_MULTI_STEP_PROPOSAL_IN_EXECUTION_KEY);
+        aborts_if is_multi_step_proposal && simple_map::spec_contains_key(metadata, is_multi_step_in_execution_key);
+
+        let post post_voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
+        let post post_metadata = table::spec_get(post_voting_forum.proposals, proposal_id).metadata;
+        ensures post_voting_forum.next_proposal_id == voting_forum.next_proposal_id + 1;
+        // property 1: Ensure that newly created proposals exist in the voting forum proposals table.
+        ensures table::spec_contains(post_voting_forum.proposals, proposal_id);
+        ensures if (is_multi_step_proposal) {
+            simple_map::spec_get(post_metadata, is_multi_step_in_execution_key) == std::bcs::serialize(false)
+        } else {
+            !simple_map::spec_contains_key(post_metadata, is_multi_step_in_execution_key)
+        };
+    }
+
+    spec get_resolution_time_secs {
+        include AbortsIfNotContainProposalID<ProposalType>;
+        let voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
+        let proposal = table::spec_get(voting_forum.proposals, proposal_id);
+        ensures result == proposal.resolution_time_secs;
+    }
+
+    spec is_multi_step_proposal_in_execution {
+        include AbortsIfNotContainProposalID<ProposalType>;
+        let voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
+        let proposal = table::spec_get(voting_forum.proposals,proposal_id);
+        aborts_if !std::string::spec_internal_check_utf8(IS_MULTI_STEP_PROPOSAL_IN_EXECUTION_KEY);
+
+        let execution_key = std::string::spec_utf8(IS_MULTI_STEP_PROPOSAL_IN_EXECUTION_KEY);
+        aborts_if !simple_map::spec_contains_key(proposal.metadata,execution_key);
+
+        let is_multi_step_in_execution_key = simple_map::spec_get(proposal.metadata,execution_key);
+        aborts_if !from_bcs::deserializable<bool>(is_multi_step_in_execution_key);
+
+        ensures result == from_bcs::deserialize<bool>(is_multi_step_in_execution_key);
+    }
+
+    spec is_voting_period_over {
+        // use supra_framework::chain_status;
+        // requires chain_status::is_operating();
+        // aborts_if false;
+        ensures result == (timestamp::spec_now_seconds() > proposal.expiration_secs);
+    }
+
+    spec fun spec_get_proposal_expiration_secs<ProposalType: store>(
+        voting_forum_address: address,
+        proposal_id: u64,
+    ): u64 {
+        let voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
```

Comment:
Don't we need `aborts_if` if `proposal_id` doesn't exist? Perhaps some earlier schema that has been used can be reused here?

Reply:
This is a spec function. aborts_if should be specified in the spec for actual function.